### DISCUSSION TITLE
Signal system consider close rails as a single signal block

### DISCUSSION THREAD

**Discussion by AntoineXYZ** [voxeltycoon/issues#591] [General]
**Describe the bug**
If 2 rail tracks are too close to each other the signal system seems to consider the tracks as a single signal block instead of two separate

**Steps to reproduce**
Build tracks like the left side, put a train inside, look at the signals
![Image of the bug](https://i.imgur.com/I4wUJff.png)
**Build version**
0.71 2019-06-26 ITCH PRE-ALPHA

**Platform**
Windows 10

**Reproducibility**
Every time.

**Comment by Garagoth** [comment]
Example with a train on tracks.
Took me a while to find the problem, pretty confusing!

(0.85.0.24 steam on Linux)

![image](https://user-images.githubusercontent.com/916161/114952240-2b8d7900-9e56-11eb-979b-1a43c1f1d0bc.png)

**Comment by IDragonfire** [comment]
Rails do not have to be continuous

Actual red signal because of "close" rail:
![20210419220203_1](https://user-images.githubusercontent.com/1217999/115297219-40bb1e00-a15c-11eb-91f0-5a3aa0c9aa2a.jpg)

If you remove a bit, it gets greeen:
![20210419220207_1](https://user-images.githubusercontent.com/1217999/115297197-3862e300-a15c-11eb-830b-d67485b6e226.jpg)

Source:
https://steamcommunity.com/linkfilter/?url=https://github.com/voxeltycoon/bug-tracker/issues/63

(0.85.0.24 steam on win10)

**Comment by katepolak** [comment]
Just to give a heads up for anyone looking at this, the is unlikely to be fixed until the big road and rail network mentioned in the post-launch FAQ: https://voxeltycoon.xyz/devlog/ea-faq/#i-have-hard-times-building-bridges-connecting-roads-etc

**Comment by victorleblais** [comment]
Also, some specific situations (like almost all the ones visible in the screenshots above) will remain: trains would collide if they were able to go on these tracks.
